packages/insomnia/src/templating/utils.ts

```diff
@@ -83,7 +83,9 @@ export function forceBracketNotation(prefix: string, key: string | number) {
 }
 
 export function normalizeToDotAndBracketNotation(prefix: string) {
-  return objectPath.normalize(prefix);
+  // objectPath.normalize will enter endless loop if last char in key name is backslash, just remove backslashes at the end of key name
+  // workaround for https://github.com/Kong/insomnia/issues/7286
+  return objectPath.normalize(prefix.replace(/\\+']/g, '\']'));
```

Comment:
Maybe we remove the backslash entirely, to prevent both the app freeze as well as the dangling environment variable which we then cannot reference in a request.
```suggestion
  return objectPath.normalize(prefix.replace(/\\+']/g, '']'));
```